### ISSUE TITLE
semanticTokens/full/delta not working

### ISSUE THREAD

**Issue by moosq** [microsoft/vscode-languageserver-node#1074]
I am using 
Version: 1.71.0 (user setup)
Commit: 784b0177c56c607789f9638da7b6bf3230d47a8c
Date: 2022-09-01T07:36:10.600Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19042
Sandboxed: No

and following TS modules for client support:
vscode-languageclient: "8.0.2"

What I try to achieve:
I have a very large file which takes several seconds to generate the semantic tokens for the semanticTokens/full Response. I would like to split it into several ranges to keep the system responsive. To achieve this I append `resultId`field to the Result message. Only the desired `semanticTokens/full/delta` request does not appear.  

```
[Trace - 6:12:15 PM] Received response 'textDocument/semanticTokens/full - (10)' in 1244ms.
Result: {
    "resultId": "someString",
    "data": [
        8,
        7,
        12,
        0,
```

The server correctly advertises its semanticTokens capabilities:
```
Received response 'initialize - (0)' in 6ms.
Result: {
    "capabilities": {
        "textDocumentSync": {
            "openClose": true,
            "change": 1,
            "save": {}
        },
        "completionProvider": {
            "triggerCharacters": [
                "."
            ]
        },
        "hoverProvider": true,
        "signatureHelpProvider": {},
        "definitionProvider": true,
        "typeDefinitionProvider": false,
        "implementationProvider": false,
        "referencesProvider": true,
        "documentSymbolProvider": true,
        "codeActionProvider": false,
        "codeLensProvider": {},
        "documentLinkProvider": {},
        "documentOnTypeFormattingProvider": {
            "firstTriggerCharacter": ""
        },
        "foldingRangeProvider": false,
        "executeCommandProvider": {
            "commands": null
        },
        "semanticTokensProvider": {
            "documentSelector": [
                {
                    "language": "ttcn3",
                    "scheme": "file",
                    "pattern": "**/*.ttcn3"
                }
            ],
            "legend": {
                "tokenTypes": [
                    "namespace",
                    "type",
                    "class",
                    "enum",
                    "interface",
                    "struct",
                    "typeParameter",
                    "parameter",
                    "variable",
                    "property",
                    "enumMember",
                    "event",
                    "function",
                    "method",
                    "macro",
                    "keyword",
                    "modifier",
                    "comment",
                    "string",
                    "number",
                    "regexp",
                    "operator"
                ],
                "tokenModifiers": [
                    "undefined",
                    "declaration",
                    "definition",
                    "readonly",
                    "static",
                    "deprecated",
                    "abstract",
                    "async",
                    "modification",
                    "documentation",
                    "defaultLibrary"
                ]
            },
            "range": true,
            "full": {
                "delta": true
            }
        },
        "workspace": {
            "workspaceFolders": {
                "supported": true,
                "changeNotifications": "workspace/didChangeWorkspaceFolders"
            }
        }
    },
```
What already works:
* semanticTokens/full
* semanticTokens/range

Is there something I miss in understanding how the interaction between server and client should be or does the lsp client just not react correctly?

**Comment by dbaeumer** [comment]
The delta request is sent only after a full request has successfully be return and the user types in the document. The server is then ask to send a delta which will move a full result FR1 into a FR2. 

What what you describe I think you don't want to have a full request at all and only provide range results. Hover this might cause flickering in the UI and will lead to a uncolored mini map.

I will close the issue. Please ping if things are still unclear.

**Comment by moosq** [comment]
@dbaeumer Thanks for clarification. I still have some more general questions:
* does the typescript lsp client library I mentioned above block the whole lsp client until it receives the `semanticTokens/full` Response or is it capable to issue other requests in the meantime
* is it possible to cancel the `semanticTokens/full` Request without displaying this event in the UI. Further on it should not lead to further `semanticTokens/full` Request being issued by the client.

**Comment by dbaeumer** [comment]
> does the typescript lsp client library I mentioned above block the whole lsp client until it receives the semanticTokens/full Response or is it capable to issue other requests in the meantime

No. LSP allows parallel requests. It is up to the server to decide what to do with them

> is it possible to cancel the semanticTokens/full Request without displaying this event in the UI. Further on it should not lead to further semanticTokens/full Request being issued by the client.

It is possible for a server to cancel a request. However the client will re-issue the request after a timeout. If you want a server to not respond to these requests anymore use dynamic registration and un-registration.
